### PR TITLE
Add shortcut for creating `Format[E#Value]` for any Enum `E`.

### PR DESCRIPTION
**Pull request by aaabramov**
# Pull Request Checklist

* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Typesafe CLA](https://www.typesafe.com/contribute/cla)?
* [x] Have you [squashed your commits](https://www.playframework.com/documentation/latest/WorkingWithGit#Squashing-commits)?
* [x] Have you added copyright headers to new files?
* [x] Have you updated the documentation?
* [x] Have you added tests for any changed functionality?

## Purpose

What does this PR do?

Add shortcut for creating `Format[E#Value]` for any Enum `E`.

### CODE REVIEW

**Comment by Abrasha** [comment]
> To fix, format your sources using sbt scalariformFormat test:scalariformFormat before submitting a pull request.

Got it. Will fix in few days

File: play-json/shared/src/test/scala/play/api/libs/json/FormatEnumSpec.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+class FormatEnumSpec extends WordSpec with MustMatchers {
+
+  import EnumWithCustomNames._
+  import EnumWithDefaultNames._
+
+  "EnumFormat" should {
+
+    "serialize correctly enum with custom names" in {
+
+      toJson(customEnum1).toString mustEqual """"ENUM1""""
```
[the remark this answers is not in the document]
Comment (andrii-abramov) [reply play-json/shared/src/test/scala/play/api/libs/json/FormatEnumSpec.scala]: @cchantep actually there is. Or using escaping `"\"ENUM1\""`

[the remark this answers is not in the document]
Comment (cchantep) [reply play-json/shared/src/test/scala/play/api/libs/json/FormatEnumSpec.scala]: I personnally don't expect the names to be quoted

[the remark this answers is not in the document]
Comment (andrii-abramov) [reply play-json/shared/src/test/scala/play/api/libs/json/FormatEnumSpec.scala]: @cchantep they are not. `Json.toString` does the quoting to produce valid JSON.

[the remark this answers is not in the document]
Comment (cchantep) [reply play-json/shared/src/test/scala/play/api/libs/json/FormatEnumSpec.scala]: Then rather `mustEqual JsString("ENUM1")`

**Comment by andrii-abramov** [comment]
Hi @cchantep, I have updated my PR according to your suggestions. Could you please review one more time?

File: play-json/shared/src/test/scala/play/api/libs/json/FormatEnumSpec.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    "serialize correctly enum with default names" in {
+
+      toJson(defaultEnum1) mustEqual JsString("defaultEnum1")
+      toJson(defaultEnum2) mustEqual JsString("defaultEnum2")
+
+    }
+
+    "deserialize correctly enum with custom names" in {
+
+      parse(""""ENUM1"""").validate[EnumWithCustomNames] mustEqual JsSuccess(customEnum1)
```
Comment (cchantep) [inline play-json/shared/src/test/scala/play/api/libs/json/FormatEnumSpec.scala]: `JsString("ENUM1").validate`

File: play-json/shared/src/test/scala/play/api/libs/json/FormatEnumSpec.scala [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      parse(""""ENUM2"""").validate[EnumWithCustomNames] mustEqual JsSuccess(customEnum2)
+
+    }
+
+    "deserialize correctly enum with default names" in {
+
+      parse(""""defaultEnum1"""").validate[EnumWithDefaultNames] mustEqual JsSuccess(defaultEnum1)
```
Comment (cchantep) [inline play-json/shared/src/test/scala/play/api/libs/json/FormatEnumSpec.scala]: same

**Comment by lightbend-cla-validator** [comment]
Hi @aaabramov,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: play-json/shared/src/test/scala/play/api/libs/json/FormatEnumSpec.scala (hunk 2 above)
Comment (aaabramov) [reply play-json/shared/src/test/scala/play/api/libs/json/FormatEnumSpec.scala]: @cchantep done

**Comment by aaabramov** [comment]
@cchantep any update on this? I would like to see this in the next release

**Comment by cchantep** [comment]
There is still one pending comment https://github.com/playframework/play-json/pull/140/files#r179757400

File: play-json/shared/src/test/scala/play/api/libs/json/FormatEnumReadsSpec.scala [hunk 4]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.api.libs.json
+
+import org.scalatest._
+
+
+class FormatEnumReadsSpec extends WordSpec with MustMatchers {
```
Comment (cchantep) [inline play-json/shared/src/test/scala/play/api/libs/json/FormatEnumReadsSpec.scala]: Seems I was clear enough. I mean that it should either be the initial code for a `FormatSharedSpec` for all the tests about `Format`, or, what I would prefer, be put in the existing `ReadSharedSpec` and `WriteSharedSpec`.

I don't see there need/benefit to separate those tests.

Comment (aaabramov) [reply play-json/shared/src/test/scala/play/api/libs/json/FormatEnumReadsSpec.scala]: @cchantep from my point of view, there is no need to put them together.
If you would like to, you can update this PR

Comment (cchantep) [reply play-json/shared/src/test/scala/play/api/libs/json/FormatEnumReadsSpec.scala]: For the few amount of code, there is no need to introduce more files, and make the tests more complex

Comment (aaabramov) [reply play-json/shared/src/test/scala/play/api/libs/json/FormatEnumReadsSpec.scala]: @cchantep done, I hope that TestEnums could stay there and we won't duplicate them (:

**Comment by aaabramov** [comment]
Hi @cchantep. Could you advice what is an estimate for the next release?

**Pull request merged**